
1). The given dataset contains price of second-hand Hyundai grand i10 car with
respect to year of making Find the best linear relationship between year and
price  Can you predict the possible price of a 2022 model second-hand grand i10?
Please learn about lasso regression and create a model along with linear
regression. Find out which one is performing better.

In [35]:
import pandas as pd
data=pd.read_csv("/home/hari/Downloads/carage.csv")

In [34]:
data.head()

Year   Price
0  2018  465000
1  2019  755000
2  2019  700000
3  2018  465000
4  2018  465000

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112 entries, 0 to 111
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Year    112 non-null    int64
 1   Price   112 non-null    int64
dtypes: int64(2)
memory usage: 1.9 KB


In [36]:
data.describe()

Year          Price
count   112.000000     112.000000
mean   2016.669643  483866.044643
std       1.629616   91217.450533
min    2013.000000  300000.000000
25%    2015.000000  423750.000000
50%    2017.000000  500000.000000
75%    2017.000000  550000.000000
max    2020.000000  755000.000000

In [10]:
data["Year"].value_counts()

2017    47
2015    22
2019    12
2018    11
2016     6
2014     6
2020     4
2013     4
Name: Year, dtype: int64

In [11]:
duplicated=data[data.duplicated()]

In [12]:
duplicated

Year   Price
3    2018  465000
4    2018  465000
9    2017  575000
15   2017  500000
16   2017  500000
18   2019  500000
19   2017  500000
20   2018  500000
21   2019  500000
23   2017  550000
24   2017  550000
25   2017  550000
26   2017  550000
27   2017  550000
28   2017  550000
29   2017  550000
31   2017  550000
32   2017  550000
33   2017  550000
35   2017  550000
36   2017  550000
41   2019  600000
44   2019  550000
46   2019  500000
47   2018  450000
51   2018  550000
53   2013  345000
55   2017  550000
57   2017  550000
62   2017  540000
64   2015  490000
66   2017  525000
72   2015  300000
74   2017  490000
75   2017  490000
76   2015  450000
79   2015  400000
82   2015  390000
84   2015  325000
87   2017  520000
88   2017  520000
89   2017  520000
90   2017  520000
91   2017  520000
92   2017  520000
93   2017  520000
94   2017  520000
95   2017  520000
96   2017  520000
101  2015  400000
102  2015  350000
105  2014  400000
106  2015  300000
111  2015  425000

In [13]:
data.drop_duplicates(inplace=True)

In [14]:
data.shape

(58, 2)

In [15]:
data["Year"].value_counts()

2017    15
2015    12
2019     7
2018     6
2016     6
2014     5
2020     4
2013     3
Name: Year, dtype: int64

In [16]:
data.isna().sum()

Year     0
Price    0
dtype: int64

### Linear Regression Model

In [17]:
x=data.drop(["Price"],axis=1)

In [18]:
x

Year
0    2018
1    2019
2    2019
5    2015
6    2016
7    2017
8    2017
10   2019
11   2018
12   2017
13   2019
14   2017
17   2017
22   2015
30   2019
34   2017
37   2017
38   2018
39   2020
40   2019
42   2018
43   2019
45   2018
48   2016
49   2018
50   2013
52   2013
54   2017
56   2014
58   2014
59   2016
60   2015
61   2017
63   2017
65   2017
67   2015
68   2015
69   2015
70   2015
71   2017
73   2013
77   2015
78   2015
80   2015
81   2014
83   2017
85   2015
86   2017
97   2020
98   2020
99   2020
100  2016
103  2017
104  2014
107  2016
108  2014
109  2015
110  2016

In [19]:
y=data["Price"]
y

0      465000
1      755000
2      700000
5      350000
6      425000
7      575000
8      509999
10     500000
11     500000
12     500000
13     600000
14     475000
17     550000
22     434999
30     650000
34     450000
37     486000
38     475000
39     545000
40     525000
42     550000
43     550000
45     450000
48     396000
49     600000
50     325000
52     345000
54     525000
56     341000
58     450000
59     550000
60     490000
61     540000
63     595000
65     495000
67     400000
68     465000
69     300000
70     450000
71     490000
73     320000
77     325000
78     409999
80     390000
81     390000
83     480000
85     320000
86     520000
97     640000
98     700000
99     600000
100    350000
103    375000
104    400000
107    375000
108    300000
109    425000
110    420000
Name: Price, dtype: int64

In [20]:
from sklearn.model_selection import train_test_split

In [21]:
x_train,x_test,y_train,y_test=train_test_split(x,y,random_state=1,test_size=0.2)

In [22]:
from sklearn.linear_model import LinearRegression
lr=LinearRegression()
lr_model=lr.fit(x_train,y_train)
lr_predict=lr_model.predict(x_test)

In [23]:
from sklearn.metrics import mean_squared_error,r2_score

In [24]:
print("Mean Squared Error =",mean_squared_error(y_test,lr_predict))
print("R Squared Error=",r2_score(y_test,lr_predict))

Mean Squared Error = 3545632860.9680295
R Squared Error= 0.7258825663162266


### Lasso Regression model

In [25]:
from sklearn.linear_model import Lasso
lasso=Lasso(alpha=0.1)
ls_model=lasso.fit(x_train,y_train)
ls_predict=ls_model.predict(x_test)

In [26]:
print("Mean Squared Error=",mean_squared_error(y_test,ls_predict))
print("R Squared Error=",r2_score(y_test,ls_predict))

Mean Squared Error= 3545634157.5119324
R Squared Error= 0.7258824660787511


### Prediction of the Price in the year 2022(Using Linear Regression)

In [27]:
x_2022=pd.DataFrame({'Year':[2022]})

In [28]:
lr_predict=lr_model.predict(x_2022)

In [37]:
import numpy as np
np.round(lr_predict,2)

array([714251.93])

We get te value 714251.93 
.

The value of 2022 model seond-hand Hundai grand i10 car is 714251.93